### PR TITLE
PHP-114 - Removed anonymous unions/structs

### PR DESCRIPTION
**Pull request by mpenick**

### CODE REVIEW

File: ext/src/BatchStatement.c [hunk 1]
```diff
@@ -94,9 +94,9 @@ PHP_METHOD(BatchStatement, add)
 
 #if PHP_MAJOR_VERSION >= 7
   ZVAL_PTR(&entry, batch_statement_entry);
-  zend_hash_next_index_insert(&self->statements, &entry);
+  zend_hash_next_index_insert(&self->data.batch.statements, &entry);
 #else
-  zend_hash_next_index_insert(&self->statements,
+  zend_hash_next_index_insert(&self->data.batch.statements,
```
Comment (mikefero) [inline ext/src/BatchStatement.c:99]: Shouldn't we be using the macros here (e.g. **`PHP5TO7_ZEND_HASH_NEXT_INDEX_INSERT`**)?

File: ext/src/Set.c [hunk 2]
```diff
@@ -315,11 +315,10 @@ php_driver_set_properties(zval *object TSRMLS_DC)
   HashTable     *props = zend_std_get_properties(object TSRMLS_CC);
 
 
-  if (PHP5TO7_ZEND_HASH_UPDATE(props,
-                               "type", sizeof("type"),
-                               PHP5TO7_ZVAL_MAYBE_P(self->type), sizeof(zval))) {
+  PHP5TO7_ZEND_HASH_UPDATE(props,
+                           "type", sizeof("type"),
+                           PHP5TO7_ZVAL_MAYBE_P(self->type), sizeof(zval));
     Z_ADDREF_P(PHP5TO7_ZVAL_MAYBE_P(self->type));
```
Comment (mikefero) [inline ext/src/Set.c]: nitpick: Spacing

File: ext/src/Type/Map.c [hunk 3]
```diff
@@ -158,17 +158,15 @@ php_driver_type_map_properties(zval *object TSRMLS_DC)
   php_driver_type *self  = PHP_DRIVER_GET_TYPE(object);
   HashTable      *props = zend_std_get_properties(object TSRMLS_CC);
 
-  if (PHP5TO7_ZEND_HASH_UPDATE(props,
+  PHP5TO7_ZEND_HASH_UPDATE(props,
                                "keyType", sizeof("keyType"),
-                               PHP5TO7_ZVAL_MAYBE_P(self->key_type), sizeof(zval))) {
-    Z_ADDREF_P(PHP5TO7_ZVAL_MAYBE_P(self->key_type));
-  }
+                               PHP5TO7_ZVAL_MAYBE_P(self->data.map.key_type), sizeof(zval));
```
Comment (mikefero) [inline ext/src/Type/Map.c]: nitpick: Spacing

File: ext/src/Type/Set.c [hunk 4]
```diff
@@ -136,11 +133,10 @@ php_driver_type_set_properties(zval *object TSRMLS_DC)
   php_driver_type *self  = PHP_DRIVER_GET_TYPE(object);
   HashTable      *props = zend_std_get_properties(object TSRMLS_CC);
 
-  if (PHP5TO7_ZEND_HASH_UPDATE(props,
+  PHP5TO7_ZEND_HASH_UPDATE(props,
                                "valueType", sizeof("valueType"),
-                               PHP5TO7_ZVAL_MAYBE_P(self->value_type), sizeof(zval))) {
-    Z_ADDREF_P(PHP5TO7_ZVAL_MAYBE_P(self->value_type));
-  }
+                               PHP5TO7_ZVAL_MAYBE_P(self->data.set.value_type), sizeof(zval));
```
Comment (mikefero) [inline ext/src/Type/Set.c]: nitpick: Spacing

File: ext/src/UserTypeValue.c [hunk 5]
```diff
@@ -355,11 +353,10 @@ php_driver_user_type_value_properties(zval *object TSRMLS_DC)
   php_driver_user_type_value *self = PHP_DRIVER_GET_USER_TYPE_VALUE(object);
   HashTable                 *props = zend_std_get_properties(object TSRMLS_CC);
 
-  if (PHP5TO7_ZEND_HASH_UPDATE(props,
+  PHP5TO7_ZEND_HASH_UPDATE(props,
                                "type", sizeof("type"),
-                               PHP5TO7_ZVAL_MAYBE_P(self->type), sizeof(zval))) {
-    Z_ADDREF_P(PHP5TO7_ZVAL_MAYBE_P(self->type));
-  }
+                               PHP5TO7_ZVAL_MAYBE_P(self->type), sizeof(zval));
```
Comment (mikefero) [inline ext/src/UserTypeValue.c]: nitpick: Spacing

**Review by mikefero — APPROVED** [review]
+1; mostly nitpicks on spacing

File: ext/src/BatchStatement.c (hunk 1 above)
Comment (mpenick) [reply ext/src/BatchStatement.c:99]: `PHP5TO7_ZEND_HASH_NEXT_INDEX_INSERT()` only works for inserting `zval`. This is inserting `php_driver_batch_statement_entry`.

**Comment by mpenick** [comment]
Merged at cb1023c2525a24634357b4ca4ad44f8c89e8b1cb

**Pull request closed without merging**